### PR TITLE
Hunting Queries changes for KQL validations fix

### PR DESCRIPTION
**Pull request by v-vdixit**
Required items, please complete
   
   Change(s):
   - Updated queries to resolve the failing kql validations

   Reason for Change(s):
   - See guidance below

   Version Updated:
   - Required only for Detections/Analytic Rule templates
   - See guidance below

   Testing Completed:
   - See guidance below

   Checked that the validations are passing and have addressed any issues that are present:
   - See guidance below

# Guidance <- remove section before submitting
-----------------------------------------------------------------------------------------------------------
## Before submitting this PR please ensure that you have read the following sections and filled out the changes, reason for change and testing complete sections:

Thank you for your contribution to the Microsoft Sentinel Github repo.

> Details of the code changes in your submitted PR.  Providing descriptions for pull requests ensures there is context to changes being made and greatly enhances the code review process.  Providing associated Issues that this resolves also easily connects the reason.
   
   Change(s):
   - Updated syntax for XYZ.yaml

   Reason for Change(s):
   - New schema used for XYZ.yaml
   - Resolves ISSUE #1234

   Version updated:
   - Yes
   - Detections/Analytic Rule templates are required to have the version updated

> The code should have been tested in a Microsoft Sentinel environment that does not have any custom parsers, functions or tables, so that you validate no incorrect syntax and execution functions properly.  If your submission requires a custom parser or function, it must be submitted with the PR.

   Testing Completed:
   - Yes/No/Need Help

_Note: If updating a detection, you must update the version field._

> Before the submission has been made, please look at running the KQL and Yaml Validation Checks locally.
> https://github.com/Azure/Azure-Sentinel#run-kql-validation-locally

   Checked that the validations are passing and have addressed any issues that are present:
   - Yes/No/Need Help
   
   _Note: Let us know if you have tried fixing the validation error and need help._

> References: 
> - [Guidance for Detection checks](https://github.com/Azure/Azure-Sentinel#pull-request-detection-template-structure-validation-check)
> - [General contribution guidance](https://github.com/Azure/Azure-Sentinel/wiki#what-can-you-contribute-and-how-can-you-create-contributions)
> - [PR validation troubleshooting](https://github.com/Azure/Azure-Sentinel#pull-request)

-----------------------------------------------------------------------------------------------------------

### CODE REVIEW

File: Solutions/Endpoint Threat Protection Essentials/Hunting Queries/PersistViaIFEORegistryKey.yaml [hunk 1]
```diff
@@ -61,10 +61,10 @@ query: |
   (
   imRegistry
   | where RegistryKey has_all ("HKEY_LOCAL_MACHINE", "Image File Execution Options")
-  | summarize Count=count() by Dvc, Username, RegistryKey
+  | summarize Count=count() by Dvc, RegistryKey
   | top 10 by Count desc
-  | extend AccountCustomEntity = Username, HostCustomEntity = Dvc
```
Comment (devikamehra) [inline Solutions/Endpoint Threat Protection Essentials/Hunting Queries/PersistViaIFEORegistryKey.yaml:66]: The placement of the line was also correct. Did not get the intention of the change.
Please provide more details.

**Review by devikamehra — CHANGES_REQUESTED** [review]

File: Hunting Queries/Microsoft 365 Defender/Impact/unusual-volume-of-file-deletion.yaml [hunk 2]
```diff
@@ -43,7 +43,7 @@ query: |
     ;   
    //
   deleteEvents  
-  | join kind=leftsemi suspUsers on AccountObjectId, SessionId
+  | join kind=leftsemi riskyUsers on AccountObjectId, SessionId
```
[the remark this answers is not in the document]
Comment (v-vdixit) [reply Hunting Queries/Microsoft 365 Defender/Impact/unusual-volume-of-file-deletion.yaml:46]: As the author of the query is not from Microsoft, we don't have direct access to reach out to them for clarifications, if there is any other process for testing this, please let us know, thanks!

File: Hunting Queries/ASimProcess/imProcess_SolarWindsInventory.yaml [hunk 3]
```diff
@@ -13,6 +13,7 @@ tags:
 query: |
   imProcessCreate
   | where Process has 'solarwinds' 
+  | extend TargetProcessFilePath = column_ifexists("TaregetProcessFilePath",TargetProcessName)
   | extend MachineName = DvcHostname , Process = TargetProcessFilePath
   | summarize StartTime = min(TimeGenerated), EndTime = max(TimeGenerated), MachineCount = dcount(Dvc), AccountCount = dcount(User), MachineNames = make_set(Dvc),
```
[the remark this answers is not in the document]
Comment (v-vdixit) [reply Hunting Queries/ASimProcess/imProcess_SolarWindsInventory.yaml:18]: Can you please give us an idea as to what should be the value assigned for maximum limit in these cases, thanks!

File: Hunting Queries/Microsoft 365 Defender/TVM/devices_with_vuln_and_users_received_payload.yaml [hunk 4]
```diff
@@ -29,7 +29,7 @@ query: |
   let all_email_addresses_aff_users=
   IdentityInfo
   | where tolower(AccountName) in (all_affected_users)
-  | summarize makelist(tolower(EmailAddress));
+  | summarize makelist(tolower(MailAddress));
```
[the remark this answers is not in the document]
Comment (v-vdixit) [reply Hunting Queries/Microsoft 365 Defender/TVM/devices_with_vuln_and_users_received_payload.yaml:32]: Can you please give us an idea as to what should be the value assigned for maximum limit in this case, thanks!

File: Solutions/Endpoint Threat Protection Essentials/Hunting Queries/PersistViaIFEORegistryKey.yaml [hunk 5]
```diff
@@ -61,10 +61,10 @@ query: |
   (
   imRegistry
   | where RegistryKey has_all ("HKEY_LOCAL_MACHINE", "Image File Execution Options")
-  | summarize Count=count() by Dvc, Username, RegistryKey
+  | summarize Count=count() by Dvc, RegistryKey
```
[the remark this answers is not in the document]
Comment (v-vdixit) [reply Solutions/Endpoint Threat Protection Essentials/Hunting Queries/PersistViaIFEORegistryKey.yaml:64]: We have KQL validation failing in this with the following error message: 
Template Id: f82c89fa-c969-4d12-832f-04d55d14522c is not valid in Line: 39 col: 36
                    Errors: The name 'Username' does not refer to any known column, table, variable or function., Code: 'KS142'

File: Hunting Queries/ASimProcess/imProcess_powershell_downloads.yaml [hunk 6]
```diff
@@ -9,9 +9,10 @@ tactics:
 query: |
   imProcessCreate 
     | where Process has_any ("powershell.exe", "powershell_ise.exe") // perfromance pre-filtering
+    | extend TargetProcessFilePath = column_ifexists("TargetProcessFilePath",TargetProcessName)
     | extend FileName=tostring(split(TargetProcessFilePath, '\\')[-1])
     | where FileName in~ ("powershell.exe", "powershell_ise.exe")
     | where CommandLine has_any ("Net.WebClient", "DownloadFile", "Invoke-WebRequest", "Invoke-Shellcode", "http:")
-    | project TimeGenerated, Dvc, User, InitiatingProcessFileName, FileName, CommandLine, EventVendor, EventProduct
```
[the remark this answers is not in the document]
Comment (v-vdixit) [reply Hunting Queries/ASimProcess/imProcess_powershell_downloads.yaml:15]: ActingProcessFileName does not exist for this schema so added ActingProcessName

File: Hunting Queries/MultipleDataSources/PossibleCommandInjectionagainstAzureIR.yaml [hunk 7]
```diff
@@ -36,7 +36,7 @@ query: |
   | where ParentProcessName endswith "diawp.exe" or ParentProcessName endswith "ReportingServicesService.exe" or ParentProcessName endswith "RSPortal.exe" or ParentProcessName endswith "RsPowerBI.exe" or ParentProcessName endswith "taskexecutor.exe"
   | where ActingProcessName == "cmd.exe"
   | where (CommandLine has_any (cmdline_tokens))
-  | extend timestamp = TimeGenerated, HostCustomEntity = DvcHostname , AccountCustomEntity = ActorUsername, ProcessCustomEntity = TargetProcessFilePath
```
[the remark this answers is not in the document]
Comment (v-vdixit) [reply Hunting Queries/MultipleDataSources/PossibleCommandInjectionagainstAzureIR.yaml:39]: FolderPath was not a part of the schema so added TargetProcessName according to your other comment

File: Solutions/Endpoint Threat Protection Essentials/Hunting Queries/PersistViaIFEORegistryKey.yaml (hunk 1 above)
Comment (v-vdixit) [reply Solutions/Endpoint Threat Protection Essentials/Hunting Queries/PersistViaIFEORegistryKey.yaml:66]: We have KQL validation failing in this with the following error message: 
Template Id: f82c89fa-c969-4d12-832f-04d55d14522c is not valid in Line: 39 col: 36
Errors: The name 'Username' does not refer to any known column, table, variable or function., Code: 'KS142'

File: Hunting Queries/ASimProcess/imProcess_ProcessEntropy.yaml [hunk 8]
```diff
@@ -25,10 +24,14 @@ query: |
   let maxLimit = 6*24;
   let removeHigh = imProcessCreate 
   | where TimeGenerated >= ago(1d)
+  | extend TargetProcessFilePath = column_ifexists("TaregetProcessFilePath",TargetProcessName)
```
Comment (devikamehra) [inline Hunting Queries/ASimProcess/imProcess_ProcessEntropy.yaml:27]: imProcessCreate is a fixed schema and thus TargetProcessFilePath will never exist. extend TargetProcessFilePath to TargetProcessName only.
Moreover the spelling is incorrect so even if it was present, values will not get fetched.

File: Hunting Queries/Microsoft 365 Defender/Impact/unusual-volume-of-file-deletion.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (devikamehra) [reply Hunting Queries/Microsoft 365 Defender/Impact/unusual-volume-of-file-deletion.yaml:46]: Check with Prateek on this and my suggestion is to migrate this to AADSignIn then.

File: Solutions/Azure SQL Database solution for sentinel/Hunting Queries/HuntingQuery-BooleanBlindSQLi.yaml [hunk 9]
```diff
@@ -54,7 +54,7 @@ query: |
   | union quoteData
   | where left == right
   | extend alertText = strcat(left, "=", right)
-  | summarize AlertText=make_list(alertText, 10000) by TimeGenerated, Database, ClientIp, PrincipalName, Statement, ApplicationName, ResourceId
+  | summarize AlertText=make_list(alertText, 10000) by TimeGenerated, Database, ClientIp, PrincipalName, Statement, ApplicationName, ResourceId, HostName
```
Comment (devikamehra) [inline Solutions/Azure SQL Database solution for sentinel/Hunting Queries/HuntingQuery-BooleanBlindSQLi.yaml:57]: Please check with Manish on this query if this change is required or not.

File: Solutions/Endpoint Threat Protection Essentials/Hunting Queries/PersistViaIFEORegistryKey.yaml (hunk 5 above)
[the remark this answers is not in the document]
Comment (devikamehra) [reply Solutions/Endpoint Threat Protection Essentials/Hunting Queries/PersistViaIFEORegistryKey.yaml:64]: Check in schema file of imRegistry if the same is there or not. If not, it is required to be added in the schema file then.

File: Solutions/Legacy IOC based Threat Protection/Hunting Queries/Dev-0322CommandLineActivityNovember2021(ASIMVersion).yaml [hunk 10]
```diff
@@ -38,19 +38,26 @@ query: |
     // Create aggregate risk score
     | extend AlertRiskScore = iif(isempty(AlertRiskScore), 0.0 , AlertRiskScore)
     | extend RiskScore = RiskScore + AlertRiskScore
-    | project-reorder  FirstSeen, LastSeen, RiskScore, Dvc, DvcId, CommandLine, AccountName
+    | project-reorder  FirstSeen, LastSeen, RiskScore, Dvc, DvcId, CommandLine, User
     | extend File = split(Process, "\\")[-1]
-    | extend timestamp = FirstSeen, AccountCustomEntity = AccountName, HostCustomEntity = Dvc
+    | extend timestamp = FirstSeen
+    | extend NTDomain = tostring(split(User,'\\',0)[0]), Name = tostring(split(User,'\\',1)[0])
+    | extend HostName = tostring(split(Dvc, '.', 0)[0]), DnsDomain = tostring(strcat_array(array_slice(split(Dvc, '.'), 1, -1), '.'))
```
Comment (devikamehra) [inline Solutions/Legacy IOC based Threat Protection/Hunting Queries/Dev-0322CommandLineActivityNovember2021(ASIMVersion).yaml:45]: Extend statements are missing.
Add wherever they are missing.

**Review by devikamehra — CHANGES_REQUESTED** [review]

File: Solutions/Azure SQL Database solution for sentinel/Hunting Queries/HuntingQuery-BooleanBlindSQLi.yaml (hunk 9 above)
Comment (manishkumar1991) [reply Solutions/Azure SQL Database solution for sentinel/Hunting Queries/HuntingQuery-BooleanBlindSQLi.yaml:57]: As I can see that Query has hostname mapping, but it's not visible in result, so adding hostname in result is required.

File: Hunting Queries/ASimProcess/imProcess_ProcessEntropy.yaml [hunk 11]
```diff
@@ -115,10 +118,10 @@ query: |
...
   TargetProcessFileName, TargetProcessFilePath, TargetProcessCommandLine, ActingProcessFileName, AllHostsProcessCount, ProcessCountOnHost, DistinctHostsProcessCount, _ResourceId, DvcId , EventVendor, EventProduct
-  | project-reorder StartTime, EndTime, ResultCount, EventID, EventVendor, EventProduct, DvcHostname, ActorUserId, Account, AccountType, Weight, ProcessEntropy,TargetProcessFileName, TargetProcessFilePath, TargetProcessCommandLine, ActingProcessFileName, AllHostsProcessCount, ProcessCountOnHost, DistinctHostsProcessCount, _ResourceId, DvcId
+  | project-reorder StartTime, EndTime, ResultCount, EventOriginalUid, EventVendor, EventProduct, DvcHostname, User, ActorUserId, Weight, ProcessEntropy,TargetProcessFileName, TargetProcessFilePath, TargetProcessCommandLine, ActingProcessFileName, AllHostsProcessCount, ProcessCountOnHost, DistinctHostsProcessCount, _ResourceId, DvcId
   | sort by Weight asc, ProcessEntropy asc, TargetProcessFilePath asc 
-  | extend timestamp = StartTime, HostCustomEntity = DvcHostname, AccountCustomEntity = Account
+  | extend timestamp = StartTime
```
Comment (devikamehra) [inline Hunting Queries/ASimProcess/imProcess_ProcessEntropy.yaml:127]: Entity mapping is missing

File: Hunting Queries/Microsoft 365 Defender/Impact/unusual-volume-of-file-deletion.yaml [hunk 12]
```diff
@@ -43,7 +43,7 @@ query: |
...
   | summarize Count=count() , (Timestamp, ReportId)=arg_min(Timestamp, ReportId) by AccountObjectId, AccountDisplayName, ApplicationId, ActionType, Time=bin(Timestamp, 5m)
   // look for only those scoped users who have generated an increase in file deletion activity.
   | summarize TotalCount= countif(Count > 50), (Timestamp, ReportId)=arg_min(Timestamp, ReportId) by AccountObjectId, AccountDisplayName, ApplicationId
```
Comment (devikamehra) [inline Hunting Queries/Microsoft 365 Defender/Impact/unusual-volume-of-file-deletion.yaml:49]: Entity mapping extend statements are missing.
FullName needs to be replaced

File: Hunting Queries/MultipleDataSources/PossibleCommandInjectionagainstAzureIR.yaml [hunk 13]
```diff
@@ -36,7 +36,7 @@ query: |
...
+  | extend timestamp = TimeGenerated, HostCustomEntity = DvcHostname , AccountCustomEntity = ActorUsername, ProcessCustomEntity = TargetProcessName
   ),
   (SecurityEvent
   | where EventID == '4688'
```
Comment (devikamehra) [inline Hunting Queries/MultipleDataSources/PossibleCommandInjectionagainstAzureIR.yaml:42]: Again, old way of mapping and extend statements are required.

File: Solutions/Legacy IOC based Threat Protection/Hunting Queries/NylonTyphoonCommandLineActivity-Nov2021.yaml [hunk 14]
```diff
@@ -32,14 +32,14 @@ query: |
...
     | project-reorder  TimeGenerated, DeviceName, DeviceId, ProcessCommandLine, AccountName
     | extend timestamp = TimeGenerated, AccountCustomEntity = AccountName, HostCustomEntity = DeviceName,  ProcessCustomEntity = InitiatingProcessFileName
     ),
     (imProcessCreate
     | where (CommandLine has_all (xcopy_tokens)) or (CommandLine has_all (archive_tokens))
-    | extend timestamp = TimeGenerated, HostCustomEntity = DvcHostname , AccountCustomEntity = ActorUsername, ProcessCustomEntity = TargetProcessFilePath
+    | extend timestamp = TimeGenerated, HostCustomEntity = DvcHostname , AccountCustomEntity = ActorUsername, ProcessCustomEntity = TargetProcessName
     ),
     (SecurityEvent
     | where EventID == '4688'
```
Comment (devikamehra) [inline Solutions/Legacy IOC based Threat Protection/Hunting Queries/NylonTyphoonCommandLineActivity-Nov2021.yaml:45]: Same issue here as well.

**Review by devikamehra — CHANGES_REQUESTED** [review]

File: Hunting Queries/ASimProcess/imProcess_Dev-0056CommandLineActivityNovember2021(ASIMVersion).yaml [hunk 15]
```diff
@@ -26,18 +26,29 @@ query: |
...
   - entityType: Host
     fieldMappings:
       - identifier: HostName
-        columnName: HostCustomEntity
+        columnName: HostName
+      - identifier: DnsDomain
+        columnName: DnsDomain
   - entityType: File
     fieldMappings:
       - identifier: Name
```
Comment (devikamehra) [inline Hunting Queries/ASimProcess/imProcess_Dev-0056CommandLineActivityNovember2021(ASIMVersion).yaml:52]: Extend statement is missing here for File entity mapping.

File: Hunting Queries/Microsoft 365 Defender/Impact/unusual-volume-of-file-deletion.yaml [hunk 16]
```diff
@@ -43,21 +43,24 @@ query: |
...
    //
   deleteEvents  
-  | join kind=leftsemi suspUsers on AccountObjectId, SessionId
+  | join kind=leftsemi riskyUsers on AccountObjectId, SessionId
   | summarize Count=count() , (Timestamp, ReportId)=arg_min(Timestamp, ReportId) by AccountObjectId, AccountDisplayName, ApplicationId, ActionType, Time=bin(Timestamp, 5m)
   // look for only those scoped users who have generated an increase in file deletion activity.
   | summarize TotalCount= countif(Count > 50), (Timestamp, ReportId)=arg_min(Timestamp, ReportId) by AccountObjectId, AccountDisplayName, ApplicationId 
   | where TotalCount >= 3
   | project AccountObjectId, AccountDisplayName, ApplicationId, TotalCount, ReportId, Timestamp
+  | extend NTDomain = tostring(split(AccountDisplayName,'\\',0)[0]), Name = tostring(split(AccountDisplayName,'\\',1)[0])
```
Comment (devikamehra) [inline Hunting Queries/Microsoft 365 Defender/Impact/unusual-volume-of-file-deletion.yaml:52]: Extend statement are not added.

File: Hunting Queries/MultipleDataSources/PossibleCommandInjectionagainstAzureIR.yaml [hunk 17]
```diff
@@ -59,15 +59,25 @@ query: |
   | extend timestamp = TimeGenerated, HostCustomEntity = Computer , AccountCustomEntity = Account, ProcessCustomEntity = NewProcessName
   )
   )
+  | extend NTDomain = tostring(split(AccountCustomEntity, '\\', 0)[0]), Name = tostring(split(AccountCustomEntity, '\\', 1)[0])
+  | extend HostName = tostring(split(HostCustomEntity, '.', 0)[0]), DnsDomain = tostring(strcat_array(array_slice(split(HostCustomEntity, '.'), 1, -1), '.'))
+  | extend Account_0_Name = Name
+  | extend Account_0_NTDomain = NTDomain
+  | extend Host_0_HostName = HostName
+  | extend Host_0_DnsDomain = DnsDomain
```
Comment (devikamehra) [inline Hunting Queries/MultipleDataSources/PossibleCommandInjectionagainstAzureIR.yaml:67]: Entity mapping is missing for Process entity.

File: Solutions/Legacy IOC based Threat Protection/Hunting Queries/Dev-0322CommandLineActivityNovember2021(ASIMVersion).yaml [hunk 18]
```diff
@@ -38,19 +38,30 @@ query: |
...
+    | extend Account_0_Name = Name
+    | extend Account_0_NTDomain = NTDomain
+    | extend Host_0_HostName = HostName
+    | extend Host_0_DnsDomain = DnsDomain
```
Comment (devikamehra) [inline Solutions/Legacy IOC based Threat Protection/Hunting Queries/Dev-0322CommandLineActivityNovember2021(ASIMVersion).yaml:49]: File entity mapping is missing.

File: Solutions/Legacy IOC based Threat Protection/Hunting Queries/NylonTyphoonCommandLineActivity-Nov2021.yaml [hunk 19]
```diff
@@ -59,15 +59,25 @@ query: |
     | extend timestamp = TimeGenerated, HostCustomEntity = Computer , AccountCustomEntity = Account, ProcessCustomEntity = NewProcessName
     )
     )
+    | extend NTDomain = tostring(split(AccountCustomEntity, '\\', 0)[0]), Name = tostring(split(AccountCustomEntity, '\\', 1)[0])
+    | extend HostName = tostring(split(HostCustomEntity, '.', 0)[0]), DnsDomain = tostring(strcat_array(array_slice(split(HostCustomEntity, '.'), 1, -1), '.'))
+    | extend Account_0_Name = Name
+    | extend Account_0_NTDomain = NTDomain
+    | extend Host_0_HostName = HostName
+    | extend Host_0_DnsDomain = DnsDomain
```
Comment (devikamehra) [inline Solutions/Legacy IOC based Threat Protection/Hunting Queries/NylonTyphoonCommandLineActivity-Nov2021.yaml:67]: Extend statements are missing for Process.

**Review by devikamehra — CHANGES_REQUESTED** [review]
Add version information in all hunting queries

**Review by devikamehra — APPROVED** [review]

**Review by v-dvedak — APPROVED** [review]

**Pull request merged**